app/models/user.rb

```diff
@@ -738,8 +738,16 @@ def self.system_avatar_template(username)
 
   def self.letter_avatar_color(username)
     username ||= ""
-    color = LetterAvatar::COLORS[Digest::MD5.hexdigest(username)[0...15].to_i(16) % LetterAvatar::COLORS.length]
-    color.map { |c| c.to_s(16).rjust(2, '0') }.join
+    colors = SiteSetting.restrict_letter_avatar_colors.present? ?
+             SiteSetting.restrict_letter_avatar_colors.split("|") :
+             LetterAvatar::COLORS
+
+    select_color(username, colors)
+  end
+
+  def self.select_color(username, available_colors)
+    color = available_colors[Digest::MD5.hexdigest(username)[0...15].to_i(16) % available_colors.length]
+    SiteSetting.restrict_letter_avatar_colors.present? ? color : color.map { |c| c.to_s(16).rjust(2, '0') }.join
```

Comment:
That line feels weird. Having `select_color` reaching out to the SiteSetting feels odd. Plus, it means we'll be testing the presence of that site setting twice every time we need a letter avatar.

I like your first solution better (now that I understand the differences between the site setting and the COLORS array)